core/src/main/scala/org/apache/spark/package.scala

```diff
@@ -41,7 +41,50 @@ package org.apache
  * level interfaces. These are subject to changes or removal in minor releases.
  */
 
+import java.util.Properties
+
+import org.apache.spark.internal.Logging
+
 package object spark {
-  // For package docs only
-  val SPARK_VERSION = "2.0.0-SNAPSHOT"
+
+  object SparkBuildInfo extends Logging {
+    val unknownProp = "<unknown>"
+    val props = new Properties()
+    val resourceStream = Thread.currentThread().getContextClassLoader.
+      getResourceAsStream("spark-version-info.properties")
+    val (spark_version: String, spark_branch: String, spark_revision: String,
+      spark_build_user: String, spark_repo_url: String, spark_build_date: String) =
+      try {
+        props.load(resourceStream)
+        (
+          props.getProperty("version", unknownProp),
+          props.getProperty("branch", unknownProp),
+          props.getProperty("revision", unknownProp),
+          props.getProperty("user", unknownProp),
+          props.getProperty("url", unknownProp),
+          props.getProperty("date", unknownProp)
+        )
+      } catch {
+        case e: Exception =>
+          logError("Unable to read spark build properties.", e);
+          throw new SparkException("Error load properties from spark-version-info.properties", e)
```

Comment:
nit: "loading"

The log also seems redundant, the exception should end up in some user-visible location already.